src/pages/index.js

```diff
@@ -34,75 +44,49 @@ const enableValidation = (settings) => {
 };
 
 function handleImageFormSubmit(name, link) {
-  addImagePopup.renderLoading(true);
-  api
-    .addNewCard({ name, link })
-    .then((newCardData) => {
-      const { name, link } = newCardData;
-      cardSection.addItem(
-        new Card(
-          newCardData,
-          "#card-template",
-          imagePopup,
-          (title, link) => imagePopup.open(title, link),
-          (cardElement, cardID) => {
-            deleteImagePopup.open(cardElement, newCardData._id);
-          },
-          api
-        ).getView()
-      );
-    })
-    .catch((err) => console.error("Error adding new card:", err));
-  addImagePopup.close();
+  function makeRequest() {
+    return api.addNewCard({ name, link }).then(renderCard);
+  }
+  handleSubmit(makeRequest, addImagePopup);
 }
 
-function handleProfileFormSubmit(name, description) {
-  editProfilePopup.renderLoading(true);
-  api
-    .updateUserInfo({ name: name, about: description })
-    .then((updatedUser) => {
+function handleProfileFormSubmit(name, about) {
+  function makeRequest() {
+    return api.updateUserInfo({ name, about }).then((updatedUser) => {
       userInfo.setUserInfo({
         name: updatedUser.name,
-        job: updatedUser.about,
+        about: updatedUser.about,
+        avatar: userInfo.getUserInfo().avatar,
       });
-      editProfilePopup.close();
-    })
-    .catch((err) => {
-      console.error("Error updating user info:", err);
     });
+  }
+  handleSubmit(makeRequest, editProfilePopup);
 }
 
 function handleAvatarSubmit(link) {
-  editAvatarPopup.renderLoading(true);
-  api
-    .updateAvatar(link)
-    .then((newAvatar) => {
-      avatarIMG.src = newAvatar.avatar;
-      editAvatarPopup.close();
-    })
-    .catch((err) => {
-      console.error("Error updating avatar:", err);
+  function makeRequest() {
+    return api.updateAvatar(link).then((newAvatar) => {
+      userInfo.setUserInfo({
+        avatar: newAvatar.avatar,
```

Comment:
* When I change the avatar the whole profile data disappears https://disk.yandex.ru/i/4ehw26tajPB5vw